torch/_functorch/partitioners.py

```diff
@@ -1650,6 +1658,9 @@ def cleanup_recompute_tags(joint_module: fx.GraphModule) -> fx.GraphModule:
                 # Solution: check whether `out` has a backward hook, and if so, intentionally save `out`
                 # in forward graph outputs. With this, we can break the above circular dependency.
                 node.meta["recompute"] = CheckpointPolicy.MUST_SAVE
+        else:
+            if any(must_recompute(user) for user in node.users):
+                node.meta["recompute"] = CheckpointPolicy.MUST_SAVE
```

Comment:
To limit the blast radius, I should probably only enable this for default_partitioner, but its morally the correct thing to do always, even for min-cut.